src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -228,12 +265,18 @@ private static boolean isValidPoke() {
         return isValid;
     }
 
+    /**
+     * Returns sequence of characters based on the last Peek by converting everything to ChangeBitTo
+     * @return CharSequence representing TRUE or FALSE, represented by ChangeBitTo, in respective places where a ? appeared in CharSequence used in last called doPeek
+     */
     private static CharSequence getPokedPattern() {
         StringBuilder newPattern = new StringBuilder();
         for(int i = 0; i < peekedPattern.length(); i++) {
             char bit = peekedPattern.charAt(i);
 
```

Comment:
Below was a found bug. This should check for a T or F value instead of a question mark, because it is receiving the returned CharSequence from the last peek!